{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "import seaborn as sns\n",
    "%matplotlib inline\n",
    "from collections import defaultdict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### BLASTn results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "blastn_dir = \"../../blast_metazoa/blastn_output_files/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['../../blast_metazoa/blastn_output_files/mollusca_vs_nematoda.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_priapulida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_arthropoda.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_annelida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_mollusca.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_nemertea.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_hemichordata.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_rotifera.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_orthonectida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_platyhelminthes.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_chordata.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_tardigrada.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_onychophora.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_onychophora.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_nematoda.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_ctenophora.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_bryozoa.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_xenacoelomorpha.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_porifera.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_echinodermata.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_phoronida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_annelida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_cnidaria.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_arthropoda.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_nemertea.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_xenacoelomorpha.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_dicyemida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_rotifera.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_hemichordata.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_priapulida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_chordata.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_placozoa.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_orthonectida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_onychophora.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_platyhelminthes.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_hemichordata.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_echinodermata.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_brachiopoda.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_orthonectida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_phoronida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_dicyemida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_ctenophora.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_tardigrada.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_placozoa.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_chordata.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_bryozoa.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_priapulida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_nematoda.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_phoronida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_brachiopoda.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_brachiopoda.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_cnidaria.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_mollusca.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_dicyemida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_acanthocephala.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_platyhelminthes.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_porifera.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_arthropoda.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_bryozoa.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_rotifera.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_xenacoelomorpha.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_nemertea.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_acanthocephala.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_placozoa.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_echinodermata.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_tardigrada.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_ctenophora.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/arthropoda_vs_mollusca.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_annelida.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_cnidaria.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/mollusca_vs_porifera.blastn',\n",
       " '../../blast_metazoa/blastn_output_files/cnidaria_vs_acanthocephala.blastn']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "blastn_files = glob.glob(blastn_dir + \"*blastn\")\n",
    "blastn_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "coord_dirs = ['../retrieve_original_coords/filtered_coords/',\n",
    "              '../../arthropoda_06_22/retrieve_original_coords/filtered_coords/',\n",
    "              '../../mollusca_06_22/retrieve_original_coords/filtered_coords/']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cnidaria\n",
      "arthropoda\n",
      "mollusca\n"
     ]
    }
   ],
   "source": [
    "phyla = ['cnidaria', 'arthropoda', 'mollusca']\n",
    "cne_count_df = pd.DataFrame(columns=['phylum', 'species', 'cne_count'])\n",
    "for i in range(3):\n",
    "    sp_count_dict = {}\n",
    "    phylum = phyla[i]\n",
    "    print(phylum)\n",
    "    coord_dir = coord_dirs[i]\n",
    "    coord_files = glob.glob(coord_dir + '*orig_coords.tsv')\n",
    "    #print(coord_files)\n",
    "    for coord_file in coord_files:\n",
    "        species = coord_file.split(\"/\")[-1].split(\"_\")[0]\n",
    "        #print(species)\n",
    "        coord_df = pd.read_csv(coord_file, sep=\"\\t\")\n",
    "        num_cne = len(coord_df)\n",
    "        sp_count_dict[species] = num_cne\n",
    "    sp_count_df = pd.DataFrame(sp_count_dict.items(), columns=['species', 'cne_count'])\n",
    "    sp_count_df['phylum'] = phylum\n",
    "    cne_count_df = pd.concat([cne_count_df, sp_count_df], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>phylum</th>\n",
       "      <th>species</th>\n",
       "      <th>cne_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>spis</td>\n",
       "      <td>110941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>hsym</td>\n",
       "      <td>6532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>aaur</td>\n",
       "      <td>2049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>dgig</td>\n",
       "      <td>3644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>chem</td>\n",
       "      <td>2352</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>ofav</td>\n",
       "      <td>43953</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>aten</td>\n",
       "      <td>5576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>mvir</td>\n",
       "      <td>2730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>hvul</td>\n",
       "      <td>2540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>adig</td>\n",
       "      <td>50811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>epal</td>\n",
       "      <td>5313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>pdam</td>\n",
       "      <td>85198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>nvec</td>\n",
       "      <td>3824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>phum</td>\n",
       "      <td>194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>lpol</td>\n",
       "      <td>1173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>ptri</td>\n",
       "      <td>8881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>amel</td>\n",
       "      <td>19700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>eaff</td>\n",
       "      <td>1289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>apis</td>\n",
       "      <td>806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>aaeg</td>\n",
       "      <td>877</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>aamp</td>\n",
       "      <td>1371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>cfel</td>\n",
       "      <td>4575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>dmel</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>anas</td>\n",
       "      <td>4661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>tcas</td>\n",
       "      <td>534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>ahyp</td>\n",
       "      <td>1078</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>pvan</td>\n",
       "      <td>7275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>cdip</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>hazt</td>\n",
       "      <td>1497</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>gocc</td>\n",
       "      <td>106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>tpal</td>\n",
       "      <td>421</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>smar</td>\n",
       "      <td>1557</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>smim</td>\n",
       "      <td>25687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>obir</td>\n",
       "      <td>19911</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>csec</td>\n",
       "      <td>2383</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>alic</td>\n",
       "      <td>1999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>cscu</td>\n",
       "      <td>18140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>cgig</td>\n",
       "      <td>24676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>obim</td>\n",
       "      <td>40836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>pcan</td>\n",
       "      <td>1164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>pmax</td>\n",
       "      <td>91747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>hruf</td>\n",
       "      <td>3468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>spha</td>\n",
       "      <td>66624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>mgal</td>\n",
       "      <td>2951</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>cvir</td>\n",
       "      <td>32653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>echl</td>\n",
       "      <td>3559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>acal</td>\n",
       "      <td>4553</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>agra</td>\n",
       "      <td>1633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>myes</td>\n",
       "      <td>96983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>bgla</td>\n",
       "      <td>3217</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>lgig</td>\n",
       "      <td>1387</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        phylum species cne_count\n",
       "0     cnidaria    spis    110941\n",
       "1     cnidaria    hsym      6532\n",
       "2     cnidaria    aaur      2049\n",
       "3     cnidaria    dgig      3644\n",
       "4     cnidaria    chem      2352\n",
       "5     cnidaria    ofav     43953\n",
       "6     cnidaria    aten      5576\n",
       "7     cnidaria    mvir      2730\n",
       "8     cnidaria    hvul      2540\n",
       "9     cnidaria    adig     50811\n",
       "10    cnidaria    epal      5313\n",
       "11    cnidaria    pdam     85198\n",
       "12    cnidaria    nvec      3824\n",
       "0   arthropoda    phum       194\n",
       "1   arthropoda    lpol      1173\n",
       "2   arthropoda    ptri      8881\n",
       "3   arthropoda    amel     19700\n",
       "4   arthropoda    eaff      1289\n",
       "5   arthropoda    apis       806\n",
       "6   arthropoda    aaeg       877\n",
       "7   arthropoda    aamp      1371\n",
       "8   arthropoda    cfel      4575\n",
       "9   arthropoda    dmel       101\n",
       "10  arthropoda    anas      4661\n",
       "11  arthropoda    tcas       534\n",
       "12  arthropoda    ahyp      1078\n",
       "13  arthropoda    pvan      7275\n",
       "14  arthropoda    cdip        93\n",
       "15  arthropoda    hazt      1497\n",
       "16  arthropoda    gocc       106\n",
       "17  arthropoda    tpal       421\n",
       "18  arthropoda    smar      1557\n",
       "19  arthropoda    smim     25687\n",
       "20  arthropoda    obir     19911\n",
       "21  arthropoda    csec      2383\n",
       "22  arthropoda    alic      1999\n",
       "23  arthropoda    cscu     18140\n",
       "0     mollusca    cgig     24676\n",
       "1     mollusca    obim     40836\n",
       "2     mollusca    pcan      1164\n",
       "3     mollusca    pmax     91747\n",
       "4     mollusca    hruf      3468\n",
       "5     mollusca    spha     66624\n",
       "6     mollusca    mgal      2951\n",
       "7     mollusca    cvir     32653\n",
       "8     mollusca    echl      3559\n",
       "9     mollusca    acal      4553\n",
       "10    mollusca    agra      1633\n",
       "11    mollusca    myes     96983\n",
       "12    mollusca    bgla      3217\n",
       "13    mollusca    lgig      1387"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cne_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'aaeg': 877,\n",
       " 'aamp': 1371,\n",
       " 'aaur': 2049,\n",
       " 'acal': 4553,\n",
       " 'adig': 50811,\n",
       " 'agra': 1633,\n",
       " 'ahyp': 1078,\n",
       " 'alic': 1999,\n",
       " 'amel': 19700,\n",
       " 'anas': 4661,\n",
       " 'apis': 806,\n",
       " 'aten': 5576,\n",
       " 'bgla': 3217,\n",
       " 'cdip': 93,\n",
       " 'cfel': 4575,\n",
       " 'cgig': 24676,\n",
       " 'chem': 2352,\n",
       " 'cscu': 18140,\n",
       " 'csec': 2383,\n",
       " 'cvir': 32653,\n",
       " 'dgig': 3644,\n",
       " 'dmel': 101,\n",
       " 'eaff': 1289,\n",
       " 'echl': 3559,\n",
       " 'epal': 5313,\n",
       " 'gocc': 106,\n",
       " 'hazt': 1497,\n",
       " 'hruf': 3468,\n",
       " 'hsym': 6532,\n",
       " 'hvul': 2540,\n",
       " 'lgig': 1387,\n",
       " 'lpol': 1173,\n",
       " 'mgal': 2951,\n",
       " 'mvir': 2730,\n",
       " 'myes': 96983,\n",
       " 'nvec': 3824,\n",
       " 'obim': 40836,\n",
       " 'obir': 19911,\n",
       " 'ofav': 43953,\n",
       " 'pcan': 1164,\n",
       " 'pdam': 85198,\n",
       " 'phum': 194,\n",
       " 'pmax': 91747,\n",
       " 'ptri': 8881,\n",
       " 'pvan': 7275,\n",
       " 'smar': 1557,\n",
       " 'smim': 25687,\n",
       " 'spha': 66624,\n",
       " 'spis': 110941,\n",
       " 'tcas': 534,\n",
       " 'tpal': 421}"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sp_count_dict"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### How many CNEs are annotated as non-coding RNAs?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def annotate_rnas(blastn_file):\n",
    "    file_df = pd.DataFrame()\n",
    "    blastn_df = pd.read_csv(blastn_file, sep=\"\\t\",\n",
    "                       names = ['cne_id', 'description', 'sseqid', 'pident', 'length', 'mismatch', 'gapopen',\n",
    "                        'qstart', 'qend', 'sstart', 'send', 'evalue', 'bitscore'])\n",
    "    all_cne_ids = list(set(blastn_df['cne_id']))\n",
    "    print(len(all_cne_ids))\n",
    "    all_rna_cnes = list(set(blastn_df[blastn_df['description'].str.contains('RNA')]['cne_id']))\n",
    "    snRNA_cnes = list(set(blastn_df[blastn_df['description'].str.contains('snRNA')]['cne_id']))\n",
    "    ncRNA_cnes = list(set(blastn_df[blastn_df['description'].str.contains('ncRNA')]['cne_id']))\n",
    "    #non_RNA_cnes = list(set(blastn_df[~blastn_df['description'].str.contains('RNA')]['cne_id']))\n",
    "    all_cne_df = pd.DataFrame({'cne_id': all_cne_ids})\n",
    "    all_rna_cne_df = pd.DataFrame({'cne_id': all_rna_cnes, 'RNA':'yes'})\n",
    "    snRNA_cne_df = pd.DataFrame({'cne_id': snRNA_cnes, 'snRNA':'yes'})\n",
    "    ncRNA_cne_df = pd.DataFrame({'cne_id': ncRNA_cnes, 'ncRNA':'yes'})\n",
    "    file_df = all_cne_df.merge(all_rna_cne_df, how='left', ).merge(snRNA_cne_df, how='left')\\\n",
    "                .merge(ncRNA_cne_df, how='left').fillna('no')\n",
    "    return(file_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13270\n"
     ]
    }
   ],
   "source": [
    "moll_chord = annotate_rnas( '../../blast_metazoa/blastn_output_files/mollusca_vs_chordata.blastn')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cne_id</th>\n",
       "      <th>RNA</th>\n",
       "      <th>snRNA</th>\n",
       "      <th>ncRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>spha_cne_33204</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>spha_cne_80016</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pmax_cne_89217</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>spha_cne_51652</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pmax_cne_17241</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13265</th>\n",
       "      <td>acal_cne_6903</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13266</th>\n",
       "      <td>spha_cne_84942</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13267</th>\n",
       "      <td>spha_cne_54755</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13268</th>\n",
       "      <td>pcan_cne_4150</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13269</th>\n",
       "      <td>obim_cne_36272</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13270 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               cne_id  RNA snRNA ncRNA\n",
       "0      spha_cne_33204  yes    no    no\n",
       "1      spha_cne_80016   no    no    no\n",
       "2      pmax_cne_89217  yes    no    no\n",
       "3      spha_cne_51652  yes    no    no\n",
       "4      pmax_cne_17241   no    no    no\n",
       "...               ...  ...   ...   ...\n",
       "13265   acal_cne_6903   no    no    no\n",
       "13266  spha_cne_84942   no    no    no\n",
       "13267  spha_cne_54755   no    no    no\n",
       "13268   pcan_cne_4150  yes    no    no\n",
       "13269  obim_cne_36272   no    no    no\n",
       "\n",
       "[13270 rows x 4 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moll_chord"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cne_id</th>\n",
       "      <th>RNA</th>\n",
       "      <th>snRNA</th>\n",
       "      <th>ncRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>spha_cne_33204</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pmax_cne_89217</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>spha_cne_51652</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>spha_cne_66343</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>obim_cne_6821</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13257</th>\n",
       "      <td>spha_cne_26123</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13259</th>\n",
       "      <td>obim_cne_17029</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13261</th>\n",
       "      <td>obim_cne_52837</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13264</th>\n",
       "      <td>spha_cne_46828</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13268</th>\n",
       "      <td>pcan_cne_4150</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>no</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6683 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               cne_id  RNA snRNA ncRNA\n",
       "0      spha_cne_33204  yes    no    no\n",
       "2      pmax_cne_89217  yes    no    no\n",
       "3      spha_cne_51652  yes    no    no\n",
       "6      spha_cne_66343  yes    no   yes\n",
       "9       obim_cne_6821  yes    no   yes\n",
       "...               ...  ...   ...   ...\n",
       "13257  spha_cne_26123  yes    no   yes\n",
       "13259  obim_cne_17029  yes    no   yes\n",
       "13261  obim_cne_52837  yes    no   yes\n",
       "13264  spha_cne_46828  yes    no    no\n",
       "13268   pcan_cne_4150  yes    no    no\n",
       "\n",
       "[6683 rows x 4 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moll_chord[moll_chord['RNA'] == 'yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cne_id</th>\n",
       "      <th>RNA</th>\n",
       "      <th>snRNA</th>\n",
       "      <th>ncRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>spha_cne_66343</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>obim_cne_6821</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>agra_cne_974</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>spha_cne_41768</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>spha_cne_68670</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13251</th>\n",
       "      <td>lgig_cne_1756</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13254</th>\n",
       "      <td>pmax_cne_18466</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13257</th>\n",
       "      <td>spha_cne_26123</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13259</th>\n",
       "      <td>obim_cne_17029</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13261</th>\n",
       "      <td>obim_cne_52837</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3582 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               cne_id  RNA snRNA ncRNA\n",
       "6      spha_cne_66343  yes    no   yes\n",
       "9       obim_cne_6821  yes    no   yes\n",
       "10       agra_cne_974  yes    no   yes\n",
       "13     spha_cne_41768  yes    no   yes\n",
       "15     spha_cne_68670  yes    no   yes\n",
       "...               ...  ...   ...   ...\n",
       "13251   lgig_cne_1756  yes   yes   yes\n",
       "13254  pmax_cne_18466  yes    no   yes\n",
       "13257  spha_cne_26123  yes    no   yes\n",
       "13259  obim_cne_17029  yes    no   yes\n",
       "13261  obim_cne_52837  yes    no   yes\n",
       "\n",
       "[3582 rows x 4 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moll_chord[moll_chord['ncRNA'] == 'yes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cne_id</th>\n",
       "      <th>RNA</th>\n",
       "      <th>snRNA</th>\n",
       "      <th>ncRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>pcan_cne_320</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>obim_cne_16134</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611</th>\n",
       "      <td>bgla_cne_10474</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>990</th>\n",
       "      <td>echl_cne_6602</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1218</th>\n",
       "      <td>obim_cne_18903</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12682</th>\n",
       "      <td>bgla_cne_7846</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12825</th>\n",
       "      <td>lgig_cne_3517</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13143</th>\n",
       "      <td>acal_cne_5564</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13165</th>\n",
       "      <td>obim_cne_9326</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13251</th>\n",
       "      <td>lgig_cne_1756</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               cne_id  RNA snRNA ncRNA\n",
       "121      pcan_cne_320  yes   yes   yes\n",
       "219    obim_cne_16134  yes   yes   yes\n",
       "611    bgla_cne_10474  yes   yes   yes\n",
       "990     echl_cne_6602  yes   yes   yes\n",
       "1218   obim_cne_18903  yes   yes   yes\n",
       "...               ...  ...   ...   ...\n",
       "12682   bgla_cne_7846  yes   yes   yes\n",
       "12825   lgig_cne_3517  yes   yes   yes\n",
       "13143   acal_cne_5564  yes   yes   yes\n",
       "13165   obim_cne_9326  yes   yes   yes\n",
       "13251   lgig_cne_1756  yes   yes   yes\n",
       "\n",
       "[122 rows x 4 columns]"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "moll_chord[moll_chord['snRNA'] == 'yes']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "#### Organize BLAST results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cne_id</th>\n",
       "      <th>description</th>\n",
       "      <th>sseqid</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>mismatch</th>\n",
       "      <th>gapopen</th>\n",
       "      <th>qstart</th>\n",
       "      <th>qend</th>\n",
       "      <th>sstart</th>\n",
       "      <th>send</th>\n",
       "      <th>evalue</th>\n",
       "      <th>bitscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate ISS534 chromosome 2</td>\n",
       "      <td>gi|1755770147|gb|CP032377.1|</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>57</td>\n",
       "      <td>159</td>\n",
       "      <td>10501195</td>\n",
       "      <td>10501093</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>91.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate TY2 chromosome 2</td>\n",
       "      <td>gi|1755770144|gb|CP032374.1|</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>57</td>\n",
       "      <td>159</td>\n",
       "      <td>10503647</td>\n",
       "      <td>10503545</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>91.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate Shisler1 chromoso...</td>\n",
       "      <td>gi|1755770141|gb|CP032371.1|</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>57</td>\n",
       "      <td>159</td>\n",
       "      <td>10481868</td>\n",
       "      <td>10481766</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>91.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Caenorhabditis briggsae isolate VX34 chromosome I</td>\n",
       "      <td>gi|2202840552|gb|CP092620.1|</td>\n",
       "      <td>94.340</td>\n",
       "      <td>53</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>72</td>\n",
       "      <td>124</td>\n",
       "      <td>3830875</td>\n",
       "      <td>3830926</td>\n",
       "      <td>3.020000e-13</td>\n",
       "      <td>80.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Caenorhabditis briggsae isolate VX34 chromosome I</td>\n",
       "      <td>gi|2202840552|gb|CP092620.1|</td>\n",
       "      <td>92.453</td>\n",
       "      <td>53</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>72</td>\n",
       "      <td>124</td>\n",
       "      <td>3824164</td>\n",
       "      <td>3824113</td>\n",
       "      <td>1.400000e-11</td>\n",
       "      <td>75.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294074</th>\n",
       "      <td>spha_cne_93162</td>\n",
       "      <td>Rhabditophanes sp. KR3021 genome assembly Rhab...</td>\n",
       "      <td>gi|687916023|emb|LK995555.1|</td>\n",
       "      <td>95.000</td>\n",
       "      <td>40</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>67</td>\n",
       "      <td>106</td>\n",
       "      <td>496068</td>\n",
       "      <td>496107</td>\n",
       "      <td>2.140000e-08</td>\n",
       "      <td>63.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294075</th>\n",
       "      <td>spha_cne_93162</td>\n",
       "      <td>Rhabditophanes sp. KR3021 genome assembly Rhab...</td>\n",
       "      <td>gi|687915983|emb|LK995699.1|</td>\n",
       "      <td>95.000</td>\n",
       "      <td>40</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>67</td>\n",
       "      <td>106</td>\n",
       "      <td>32255</td>\n",
       "      <td>32294</td>\n",
       "      <td>2.140000e-08</td>\n",
       "      <td>63.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294076</th>\n",
       "      <td>spha_cne_93162</td>\n",
       "      <td>Oscheius dolichura genome assembly, chromosome: V</td>\n",
       "      <td>gi|2211271299|emb|OW051496.1|</td>\n",
       "      <td>97.222</td>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>76</td>\n",
       "      <td>111</td>\n",
       "      <td>8896691</td>\n",
       "      <td>8896726</td>\n",
       "      <td>7.690000e-08</td>\n",
       "      <td>62.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294077</th>\n",
       "      <td>spha_cne_93162</td>\n",
       "      <td>Oscheius onirici genome assembly, chromosome: III</td>\n",
       "      <td>gi|2211271102|emb|OW051470.1|</td>\n",
       "      <td>97.222</td>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>113</td>\n",
       "      <td>9054206</td>\n",
       "      <td>9054171</td>\n",
       "      <td>7.690000e-08</td>\n",
       "      <td>62.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294078</th>\n",
       "      <td>spha_cne_93162</td>\n",
       "      <td>Oscheius onirici genome assembly, chromosome: III</td>\n",
       "      <td>gi|2211271102|emb|OW051470.1|</td>\n",
       "      <td>97.222</td>\n",
       "      <td>36</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>113</td>\n",
       "      <td>9075095</td>\n",
       "      <td>9075130</td>\n",
       "      <td>7.690000e-08</td>\n",
       "      <td>62.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>294079 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                cne_id                                        description  \\\n",
       "0        acal_cne_1263   Trichinella spiralis isolate ISS534 chromosome 2   \n",
       "1        acal_cne_1263      Trichinella spiralis isolate TY2 chromosome 2   \n",
       "2        acal_cne_1263  Trichinella spiralis isolate Shisler1 chromoso...   \n",
       "3        acal_cne_1263  Caenorhabditis briggsae isolate VX34 chromosome I   \n",
       "4        acal_cne_1263  Caenorhabditis briggsae isolate VX34 chromosome I   \n",
       "...                ...                                                ...   \n",
       "294074  spha_cne_93162  Rhabditophanes sp. KR3021 genome assembly Rhab...   \n",
       "294075  spha_cne_93162  Rhabditophanes sp. KR3021 genome assembly Rhab...   \n",
       "294076  spha_cne_93162  Oscheius dolichura genome assembly, chromosome: V   \n",
       "294077  spha_cne_93162  Oscheius onirici genome assembly, chromosome: III   \n",
       "294078  spha_cne_93162  Oscheius onirici genome assembly, chromosome: III   \n",
       "\n",
       "                               sseqid  pident  length  mismatch  gapopen  \\\n",
       "0        gi|1755770147|gb|CP032377.1|  82.857     105        14        4   \n",
       "1        gi|1755770144|gb|CP032374.1|  82.857     105        14        4   \n",
       "2        gi|1755770141|gb|CP032371.1|  82.857     105        14        4   \n",
       "3        gi|2202840552|gb|CP092620.1|  94.340      53         2        1   \n",
       "4        gi|2202840552|gb|CP092620.1|  92.453      53         3        1   \n",
       "...                               ...     ...     ...       ...      ...   \n",
       "294074   gi|687916023|emb|LK995555.1|  95.000      40         2        0   \n",
       "294075   gi|687915983|emb|LK995699.1|  95.000      40         2        0   \n",
       "294076  gi|2211271299|emb|OW051496.1|  97.222      36         1        0   \n",
       "294077  gi|2211271102|emb|OW051470.1|  97.222      36         1        0   \n",
       "294078  gi|2211271102|emb|OW051470.1|  97.222      36         1        0   \n",
       "\n",
       "        qstart  qend    sstart      send        evalue  bitscore  \n",
       "0           57   159  10501195  10501093  1.390000e-16      91.6  \n",
       "1           57   159  10503647  10503545  1.390000e-16      91.6  \n",
       "2           57   159  10481868  10481766  1.390000e-16      91.6  \n",
       "3           72   124   3830875   3830926  3.020000e-13      80.5  \n",
       "4           72   124   3824164   3824113  1.400000e-11      75.0  \n",
       "...        ...   ...       ...       ...           ...       ...  \n",
       "294074      67   106    496068    496107  2.140000e-08      63.9  \n",
       "294075      67   106     32255     32294  2.140000e-08      63.9  \n",
       "294076      76   111   8896691   8896726  7.690000e-08      62.1  \n",
       "294077      78   113   9054206   9054171  7.690000e-08      62.1  \n",
       "294078      78   113   9075095   9075130  7.690000e-08      62.1  \n",
       "\n",
       "[294079 rows x 13 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "blastn_file = blastn_files[0]\n",
    "file_df = pd.DataFrame()\n",
    "blastn_df = pd.read_csv(blastn_file, sep=\"\\t\",\n",
    "                   names = ['cne_id', 'description', 'sseqid', 'pident', 'length', 'mismatch', 'gapopen',\n",
    "                    'qstart', 'qend', 'sstart', 'send', 'evalue', 'bitscore'])\n",
    "blastn_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query phylum: mollusca Target phylum: nematoda\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/gonzalezp2/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:13: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  del sys.path[0]\n",
      "/Users/gonzalezp2/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Query phylum: cnidaria Target phylum: priapulida\n",
      "Query phylum: cnidaria Target phylum: arthropoda\n",
      "Query phylum: arthropoda Target phylum: annelida\n",
      "Query phylum: cnidaria Target phylum: mollusca\n",
      "Query phylum: arthropoda Target phylum: nemertea\n",
      "Query phylum: arthropoda Target phylum: hemichordata\n",
      "Query phylum: arthropoda Target phylum: rotifera\n",
      "Query phylum: arthropoda Target phylum: orthonectida\n",
      "Query phylum: mollusca Target phylum: platyhelminthes\n",
      "Query phylum: mollusca Target phylum: chordata\n",
      "Query phylum: mollusca Target phylum: tardigrada\n",
      "Query phylum: cnidaria Target phylum: onychophora\n",
      "Query phylum: arthropoda Target phylum: onychophora\n",
      "Query phylum: arthropoda Target phylum: nematoda\n",
      "Query phylum: mollusca Target phylum: ctenophora\n",
      "Query phylum: cnidaria Target phylum: bryozoa\n",
      "Query phylum: arthropoda Target phylum: xenacoelomorpha\n",
      "Query phylum: cnidaria Target phylum: porifera\n",
      "Query phylum: cnidaria Target phylum: echinodermata\n",
      "Query phylum: arthropoda Target phylum: phoronida\n",
      "Query phylum: mollusca Target phylum: annelida\n",
      "Query phylum: cnidaria Target phylum: cnidaria\n",
      "Query phylum: arthropoda Target phylum: arthropoda\n",
      "Query phylum: mollusca Target phylum: nemertea\n",
      "Query phylum: cnidaria Target phylum: xenacoelomorpha\n",
      "Query phylum: arthropoda Target phylum: dicyemida\n",
      "Query phylum: mollusca Target phylum: rotifera\n",
      "Query phylum: mollusca Target phylum: hemichordata\n",
      "Query phylum: arthropoda Target phylum: priapulida\n",
      "Query phylum: arthropoda Target phylum: chordata\n",
      "Query phylum: cnidaria Target phylum: placozoa\n",
      "Query phylum: mollusca Target phylum: orthonectida\n",
      "Query phylum: mollusca Target phylum: onychophora\n",
      "Query phylum: cnidaria Target phylum: platyhelminthes\n",
      "Query phylum: cnidaria Target phylum: hemichordata\n",
      "Query phylum: arthropoda Target phylum: echinodermata\n",
      "Query phylum: mollusca Target phylum: brachiopoda\n",
      "Query phylum: cnidaria Target phylum: orthonectida\n",
      "Query phylum: mollusca Target phylum: phoronida\n",
      "Query phylum: cnidaria Target phylum: dicyemida\n",
      "Query phylum: arthropoda Target phylum: ctenophora\n",
      "Query phylum: cnidaria Target phylum: tardigrada\n",
      "Query phylum: arthropoda Target phylum: placozoa\n",
      "Query phylum: cnidaria Target phylum: chordata\n",
      "Query phylum: arthropoda Target phylum: bryozoa\n",
      "Query phylum: mollusca Target phylum: priapulida\n",
      "Query phylum: cnidaria Target phylum: nematoda\n",
      "Query phylum: cnidaria Target phylum: phoronida\n",
      "Query phylum: cnidaria Target phylum: brachiopoda\n",
      "Query phylum: arthropoda Target phylum: brachiopoda\n",
      "Query phylum: arthropoda Target phylum: cnidaria\n",
      "Query phylum: mollusca Target phylum: mollusca\n",
      "Query phylum: mollusca Target phylum: dicyemida\n",
      "Query phylum: arthropoda Target phylum: acanthocephala\n",
      "Query phylum: arthropoda Target phylum: platyhelminthes\n",
      "Query phylum: arthropoda Target phylum: porifera\n",
      "Query phylum: mollusca Target phylum: arthropoda\n",
      "Query phylum: mollusca Target phylum: bryozoa\n",
      "Query phylum: cnidaria Target phylum: rotifera\n",
      "Query phylum: mollusca Target phylum: xenacoelomorpha\n",
      "Query phylum: cnidaria Target phylum: nemertea\n",
      "Query phylum: mollusca Target phylum: acanthocephala\n",
      "Query phylum: mollusca Target phylum: placozoa\n",
      "Query phylum: mollusca Target phylum: echinodermata\n",
      "Query phylum: arthropoda Target phylum: tardigrada\n",
      "Query phylum: cnidaria Target phylum: ctenophora\n",
      "Query phylum: arthropoda Target phylum: mollusca\n",
      "Query phylum: cnidaria Target phylum: annelida\n",
      "Query phylum: mollusca Target phylum: cnidaria\n",
      "Query phylum: mollusca Target phylum: porifera\n",
      "Query phylum: cnidaria Target phylum: acanthocephala\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cne_id</th>\n",
       "      <th>description</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>evalue</th>\n",
       "      <th>query_phylum</th>\n",
       "      <th>target_phylum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate ISS534 chromosome 2</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate TY2 chromosome 2</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate Shisler1 chromoso...</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Caenorhabditis briggsae isolate VX34 chromosome I</td>\n",
       "      <td>94.340</td>\n",
       "      <td>53</td>\n",
       "      <td>3.020000e-13</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Caenorhabditis briggsae isolate VX34 chromosome I</td>\n",
       "      <td>92.453</td>\n",
       "      <td>53</td>\n",
       "      <td>1.400000e-11</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Southwellina hispida mitochondrion, complete g...</td>\n",
       "      <td>72.606</td>\n",
       "      <td>449</td>\n",
       "      <td>2.510000e-32</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Leptorhynchoides thecatus isolate Withbass2 cy...</td>\n",
       "      <td>73.294</td>\n",
       "      <td>337</td>\n",
       "      <td>3.270000e-26</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Pomphorhynchus tereticollis isolate Ouche mito...</td>\n",
       "      <td>81.295</td>\n",
       "      <td>139</td>\n",
       "      <td>4.230000e-25</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Pomphorhynchus tereticollis isolate Straslund ...</td>\n",
       "      <td>80.576</td>\n",
       "      <td>139</td>\n",
       "      <td>1.970000e-23</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Pomphorhynchus tereticollis isolate Ida mitoch...</td>\n",
       "      <td>79.286</td>\n",
       "      <td>140</td>\n",
       "      <td>4.260000e-20</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41668015 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            cne_id                                        description  pident  \\\n",
       "0    acal_cne_1263   Trichinella spiralis isolate ISS534 chromosome 2  82.857   \n",
       "1    acal_cne_1263      Trichinella spiralis isolate TY2 chromosome 2  82.857   \n",
       "2    acal_cne_1263  Trichinella spiralis isolate Shisler1 chromoso...  82.857   \n",
       "3    acal_cne_1263  Caenorhabditis briggsae isolate VX34 chromosome I  94.340   \n",
       "4    acal_cne_1263  Caenorhabditis briggsae isolate VX34 chromosome I  92.453   \n",
       "..             ...                                                ...     ...   \n",
       "0   hsym_cne_10568  Southwellina hispida mitochondrion, complete g...  72.606   \n",
       "1   hsym_cne_10568  Leptorhynchoides thecatus isolate Withbass2 cy...  73.294   \n",
       "2   hsym_cne_10568  Pomphorhynchus tereticollis isolate Ouche mito...  81.295   \n",
       "3   hsym_cne_10568  Pomphorhynchus tereticollis isolate Straslund ...  80.576   \n",
       "4   hsym_cne_10568  Pomphorhynchus tereticollis isolate Ida mitoch...  79.286   \n",
       "\n",
       "   length        evalue query_phylum   target_phylum  \n",
       "0     105  1.390000e-16     mollusca        nematoda  \n",
       "1     105  1.390000e-16     mollusca        nematoda  \n",
       "2     105  1.390000e-16     mollusca        nematoda  \n",
       "3      53  3.020000e-13     mollusca        nematoda  \n",
       "4      53  1.400000e-11     mollusca        nematoda  \n",
       "..    ...           ...          ...             ...  \n",
       "0     449  2.510000e-32     cnidaria  acanthocephala  \n",
       "1     337  3.270000e-26     cnidaria  acanthocephala  \n",
       "2     139  4.230000e-25     cnidaria  acanthocephala  \n",
       "3     139  1.970000e-23     cnidaria  acanthocephala  \n",
       "4     140  4.260000e-20     cnidaria  acanthocephala  \n",
       "\n",
       "[41668015 rows x 7 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df = pd.DataFrame()\n",
    "for file in blastn_files:\n",
    "    query_phylum = file.split('/')[-1].split(\"_vs_\")[0]\n",
    "    target_phylum = file.split('/')[-1].split(\"_vs_\")[1].split(\".\")[0]\n",
    "    print(\"Query phylum:\", query_phylum, \"Target phylum:\", target_phylum)\n",
    "    file_df = pd.DataFrame()\n",
    "    blastn_df = pd.read_csv(file, sep=\"\\t\",\n",
    "                   names = ['cne_id', 'description', 'sseqid', 'pident', 'length', 'mismatch', 'gapopen',\n",
    "                    'qstart', 'qend', 'sstart', 'send', 'evalue', 'bitscore'])\n",
    "    #ll_cne_ids = list(set(blastn_df['cne_id']))\n",
    "    all_cne_df = blastn_df[['cne_id', 'description', 'pident', 'length', 'evalue']]\n",
    "    all_cne_df['query_phylum'] = query_phylum\n",
    "    all_cne_df['target_phylum'] = target_phylum\n",
    "    #cluster_df = cluster_dfs[query_phylum]\n",
    "    #rna_df = rna_df.merge(cluster_df, how='left')\n",
    "    #rna_df['cluster_id'] = query_phylum + \"_\" + rna_df['cluster_id'] \n",
    "    summary_df = pd.concat([summary_df, all_cne_df], axis=0)\n",
    "summary_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cne_id</th>\n",
       "      <th>description</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>evalue</th>\n",
       "      <th>query_phylum</th>\n",
       "      <th>target_phylum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate ISS534 chromosome 2</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate TY2 chromosome 2</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate Shisler1 chromoso...</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Caenorhabditis briggsae isolate VX34 chromosome I</td>\n",
       "      <td>94.340</td>\n",
       "      <td>53</td>\n",
       "      <td>3.020000e-13</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Caenorhabditis briggsae isolate VX34 chromosome I</td>\n",
       "      <td>92.453</td>\n",
       "      <td>53</td>\n",
       "      <td>1.400000e-11</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Southwellina hispida mitochondrion, complete g...</td>\n",
       "      <td>72.606</td>\n",
       "      <td>449</td>\n",
       "      <td>2.510000e-32</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Leptorhynchoides thecatus isolate Withbass2 cy...</td>\n",
       "      <td>73.294</td>\n",
       "      <td>337</td>\n",
       "      <td>3.270000e-26</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Pomphorhynchus tereticollis isolate Ouche mito...</td>\n",
       "      <td>81.295</td>\n",
       "      <td>139</td>\n",
       "      <td>4.230000e-25</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Pomphorhynchus tereticollis isolate Straslund ...</td>\n",
       "      <td>80.576</td>\n",
       "      <td>139</td>\n",
       "      <td>1.970000e-23</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Pomphorhynchus tereticollis isolate Ida mitoch...</td>\n",
       "      <td>79.286</td>\n",
       "      <td>140</td>\n",
       "      <td>4.260000e-20</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>41668015 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            cne_id                                        description  pident  \\\n",
       "0    acal_cne_1263   Trichinella spiralis isolate ISS534 chromosome 2  82.857   \n",
       "1    acal_cne_1263      Trichinella spiralis isolate TY2 chromosome 2  82.857   \n",
       "2    acal_cne_1263  Trichinella spiralis isolate Shisler1 chromoso...  82.857   \n",
       "3    acal_cne_1263  Caenorhabditis briggsae isolate VX34 chromosome I  94.340   \n",
       "4    acal_cne_1263  Caenorhabditis briggsae isolate VX34 chromosome I  92.453   \n",
       "..             ...                                                ...     ...   \n",
       "0   hsym_cne_10568  Southwellina hispida mitochondrion, complete g...  72.606   \n",
       "1   hsym_cne_10568  Leptorhynchoides thecatus isolate Withbass2 cy...  73.294   \n",
       "2   hsym_cne_10568  Pomphorhynchus tereticollis isolate Ouche mito...  81.295   \n",
       "3   hsym_cne_10568  Pomphorhynchus tereticollis isolate Straslund ...  80.576   \n",
       "4   hsym_cne_10568  Pomphorhynchus tereticollis isolate Ida mitoch...  79.286   \n",
       "\n",
       "   length        evalue query_phylum   target_phylum  \n",
       "0     105  1.390000e-16     mollusca        nematoda  \n",
       "1     105  1.390000e-16     mollusca        nematoda  \n",
       "2     105  1.390000e-16     mollusca        nematoda  \n",
       "3      53  3.020000e-13     mollusca        nematoda  \n",
       "4      53  1.400000e-11     mollusca        nematoda  \n",
       "..    ...           ...          ...             ...  \n",
       "0     449  2.510000e-32     cnidaria  acanthocephala  \n",
       "1     337  3.270000e-26     cnidaria  acanthocephala  \n",
       "2     139  4.230000e-25     cnidaria  acanthocephala  \n",
       "3     139  1.970000e-23     cnidaria  acanthocephala  \n",
       "4     140  4.260000e-20     cnidaria  acanthocephala  \n",
       "\n",
       "[41668015 rows x 7 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summary_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Only keep BLAST hits with >75% similarity over >50 bp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cne_id</th>\n",
       "      <th>description</th>\n",
       "      <th>pident</th>\n",
       "      <th>length</th>\n",
       "      <th>evalue</th>\n",
       "      <th>query_phylum</th>\n",
       "      <th>target_phylum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate ISS534 chromosome 2</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate TY2 chromosome 2</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Trichinella spiralis isolate Shisler1 chromoso...</td>\n",
       "      <td>82.857</td>\n",
       "      <td>105</td>\n",
       "      <td>1.390000e-16</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Caenorhabditis briggsae isolate VX34 chromosome I</td>\n",
       "      <td>94.340</td>\n",
       "      <td>53</td>\n",
       "      <td>3.020000e-13</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>acal_cne_1263</td>\n",
       "      <td>Caenorhabditis briggsae isolate VX34 chromosome I</td>\n",
       "      <td>92.453</td>\n",
       "      <td>53</td>\n",
       "      <td>1.400000e-11</td>\n",
       "      <td>mollusca</td>\n",
       "      <td>nematoda</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Southwellina hispida mitochondrion, complete g...</td>\n",
       "      <td>72.606</td>\n",
       "      <td>449</td>\n",
       "      <td>2.510000e-32</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Leptorhynchoides thecatus isolate Withbass2 cy...</td>\n",
       "      <td>73.294</td>\n",
       "      <td>337</td>\n",
       "      <td>3.270000e-26</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Pomphorhynchus tereticollis isolate Ouche mito...</td>\n",
       "      <td>81.295</td>\n",
       "      <td>139</td>\n",
       "      <td>4.230000e-25</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Pomphorhynchus tereticollis isolate Straslund ...</td>\n",
       "      <td>80.576</td>\n",
       "      <td>139</td>\n",
       "      <td>1.970000e-23</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hsym_cne_10568</td>\n",
       "      <td>Pomphorhynchus tereticollis isolate Ida mitoch...</td>\n",
       "      <td>79.286</td>\n",
       "      <td>140</td>\n",
       "      <td>4.260000e-20</td>\n",
       "      <td>cnidaria</td>\n",
       "      <td>acanthocephala</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>25262566 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            cne_id                                        description  pident  \\\n",
       "0    acal_cne_1263   Trichinella spiralis isolate ISS534 chromosome 2  82.857   \n",
       "1    acal_cne_1263      Trichinella spiralis isolate TY2 chromosome 2  82.857   \n",
       "2    acal_cne_1263  Trichinella spiralis isolate Shisler1 chromoso...  82.857   \n",
       "3    acal_cne_1263  Caenorhabditis briggsae isolate VX34 chromosome I  94.340   \n",
       "4    acal_cne_1263  Caenorhabditis briggsae isolate VX34 chromosome I  92.453   \n",
       "..             ...                                                ...     ...   \n",
       "0   hsym_cne_10568  Southwellina hispida mitochondrion, complete g...  72.606   \n",
       "1   hsym_cne_10568  Leptorhynchoides thecatus isolate Withbass2 cy...  73.294   \n",
       "2   hsym_cne_10568  Pomphorhynchus tereticollis isolate Ouche mito...  81.295   \n",
       "3   hsym_cne_10568  Pomphorhynchus tereticollis isolate Straslund ...  80.576   \n",
       "4   hsym_cne_10568  Pomphorhynchus tereticollis isolate Ida mitoch...  79.286   \n",
       "\n",
       "   length        evalue query_phylum   target_phylum  \n",
       "0     105  1.390000e-16     mollusca        nematoda  \n",
       "1     105  1.390000e-16     mollusca        nematoda  \n",
       "2     105  1.390000e-16     mollusca        nematoda  \n",
       "3      53  3.020000e-13     mollusca        nematoda  \n",
       "4      53  1.400000e-11     mollusca        nematoda  \n",
       "..    ...           ...          ...             ...  \n",
       "0     449  2.510000e-32     cnidaria  acanthocephala  \n",
       "1     337  3.270000e-26     cnidaria  acanthocephala  \n",
       "2     139  4.230000e-25     cnidaria  acanthocephala  \n",
       "3     139  1.970000e-23     cnidaria  acanthocephala  \n",
       "4     140  4.260000e-20     cnidaria  acanthocephala  \n",
       "\n",
       "[25262566 rows x 7 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_df = summary_df.query('pident>75' and 'length> 50')\n",
    "filtered_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Count how many unique CNEs have a match to each phylum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "no_dup_df = filtered_df.drop_duplicates(['cne_id', 'query_phylum', 'target_phylum'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query_phylum</th>\n",
       "      <th>phylum_cne_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>124309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>325463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>375451</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  query_phylum  phylum_cne_count\n",
       "0   arthropoda            124309\n",
       "1     cnidaria            325463\n",
       "2     mollusca            375451"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "phylum_count_df = cne_count_df[['phylum', 'cne_count']].groupby('phylum').sum().reset_index().\\\n",
    "    rename({'phylum':'query_phylum', 'cne_count':'phylum_cne_count'}, axis=1)\n",
    "phylum_count_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_hit_count_df = no_dup_df[['cne_id', 'query_phylum', 'target_phylum']].groupby(['query_phylum', 'target_phylum' ]).count().\\\n",
    "    reset_index().rename({'cne_id': 'cne_count'}, axis=1)\n",
    "blast_hit_count_df = blast_hit_count_df.merge(phylum_count_df)\n",
    "blast_hit_count_df['pct_cnes'] = 100 *blast_hit_count_df['cne_count'] / blast_hit_count_df['phylum_cne_count']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query_phylum</th>\n",
       "      <th>target_phylum</th>\n",
       "      <th>cne_count</th>\n",
       "      <th>phylum_cne_count</th>\n",
       "      <th>pct_cnes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>acanthocephala</td>\n",
       "      <td>19</td>\n",
       "      <td>124309</td>\n",
       "      <td>0.015284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>annelida</td>\n",
       "      <td>2910</td>\n",
       "      <td>124309</td>\n",
       "      <td>2.340941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>arthropoda</td>\n",
       "      <td>75380</td>\n",
       "      <td>124309</td>\n",
       "      <td>60.639214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>brachiopoda</td>\n",
       "      <td>422</td>\n",
       "      <td>124309</td>\n",
       "      <td>0.339477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>bryozoa</td>\n",
       "      <td>1902</td>\n",
       "      <td>124309</td>\n",
       "      <td>1.530058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>porifera</td>\n",
       "      <td>37</td>\n",
       "      <td>375451</td>\n",
       "      <td>0.009855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>priapulida</td>\n",
       "      <td>101</td>\n",
       "      <td>375451</td>\n",
       "      <td>0.026901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>rotifera</td>\n",
       "      <td>492</td>\n",
       "      <td>375451</td>\n",
       "      <td>0.131042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>tardigrada</td>\n",
       "      <td>6</td>\n",
       "      <td>375451</td>\n",
       "      <td>0.001598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>xenacoelomorpha</td>\n",
       "      <td>14</td>\n",
       "      <td>375451</td>\n",
       "      <td>0.003729</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   query_phylum    target_phylum  cne_count  phylum_cne_count   pct_cnes\n",
       "0    arthropoda   acanthocephala         19            124309   0.015284\n",
       "1    arthropoda         annelida       2910            124309   2.340941\n",
       "2    arthropoda       arthropoda      75380            124309  60.639214\n",
       "3    arthropoda      brachiopoda        422            124309   0.339477\n",
       "4    arthropoda          bryozoa       1902            124309   1.530058\n",
       "..          ...              ...        ...               ...        ...\n",
       "63     mollusca         porifera         37            375451   0.009855\n",
       "64     mollusca       priapulida        101            375451   0.026901\n",
       "65     mollusca         rotifera        492            375451   0.131042\n",
       "66     mollusca       tardigrada          6            375451   0.001598\n",
       "67     mollusca  xenacoelomorpha         14            375451   0.003729\n",
       "\n",
       "[68 rows x 5 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "blast_hit_count_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sort data frame to arrange heatmap according to phylogeny"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sorter = ['ctenophora', 'placozoa', 'porifera', 'cnidaria', 'xenacoelomorpha', 'priapulida', 'nematoda', \n",
    "          'arthropoda', 'tardigrada', 'onychophora', 'rotifera', 'acanthocephala', 'mollusca', 'annelida', 'nemertea',\n",
    "          'dicyemida', 'orthonectida','platyhelminthes', 'phoronida', 'brachiopoda', 'bryozoa', 'echinodermata', \n",
    "          'hemichordata', 'chordata']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>query_phylum</th>\n",
       "      <th>target_phylum</th>\n",
       "      <th>cne_count</th>\n",
       "      <th>phylum_cne_count</th>\n",
       "      <th>pct_cnes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>ctenophora</td>\n",
       "      <td>8</td>\n",
       "      <td>325463</td>\n",
       "      <td>0.002458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>ctenophora</td>\n",
       "      <td>7</td>\n",
       "      <td>124309</td>\n",
       "      <td>0.005631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>ctenophora</td>\n",
       "      <td>8</td>\n",
       "      <td>375451</td>\n",
       "      <td>0.002131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>placozoa</td>\n",
       "      <td>11</td>\n",
       "      <td>325463</td>\n",
       "      <td>0.003380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>placozoa</td>\n",
       "      <td>6</td>\n",
       "      <td>375451</td>\n",
       "      <td>0.001598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>hemichordata</td>\n",
       "      <td>70</td>\n",
       "      <td>124309</td>\n",
       "      <td>0.056311</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>hemichordata</td>\n",
       "      <td>25</td>\n",
       "      <td>325463</td>\n",
       "      <td>0.007681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>arthropoda</td>\n",
       "      <td>chordata</td>\n",
       "      <td>8564</td>\n",
       "      <td>124309</td>\n",
       "      <td>6.889284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>cnidaria</td>\n",
       "      <td>chordata</td>\n",
       "      <td>1146</td>\n",
       "      <td>325463</td>\n",
       "      <td>0.352114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>mollusca</td>\n",
       "      <td>chordata</td>\n",
       "      <td>8529</td>\n",
       "      <td>375451</td>\n",
       "      <td>2.271668</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   query_phylum target_phylum  cne_count  phylum_cne_count  pct_cnes\n",
       "31     cnidaria    ctenophora          8            325463  0.002458\n",
       "7    arthropoda    ctenophora          7            124309  0.005631\n",
       "53     mollusca    ctenophora          8            375451  0.002131\n",
       "39     cnidaria      placozoa         11            325463  0.003380\n",
       "61     mollusca      placozoa          6            375451  0.001598\n",
       "..          ...           ...        ...               ...       ...\n",
       "10   arthropoda  hemichordata         70            124309  0.056311\n",
       "34     cnidaria  hemichordata         25            325463  0.007681\n",
       "5    arthropoda      chordata       8564            124309  6.889284\n",
       "29     cnidaria      chordata       1146            325463  0.352114\n",
       "51     mollusca      chordata       8529            375451  2.271668\n",
       "\n",
       "[68 rows x 5 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "blast_hit_count_df.target_phylum = blast_hit_count_df.target_phylum.astype(\"category\")\n",
    "blast_hit_count_df.target_phylum.cat.set_categories(sorter, inplace=True)\n",
    "blast_hit_count_df = blast_hit_count_df.sort_values([\"target_phylum\"])\n",
    "blast_hit_count_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Write to file for plotting in R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "blast_hit_count_df.to_csv('blast_hit_counts.tsv', sep=\"\\t\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
